recipes_source/distributed_async_checkpoint_recipe.rst

```diff
@@ -0,0 +1,259 @@
+Asynchronous Saving with Distributed Checkpoint (DCP)
+=====================================================
+
+Checkpointing is often a bottle-neck in the critical path for distributed training workloads, incurring larger and larger costs as both model and world sizes grow.
+One excellent strategy for offsetting this cost is to checkpoint in parallel, asynchronously. Below, we expand the save example
+from the `Getting Started with Distributed Checkpoint Tutorial <https://github.com/pytorch/tutorials/blob/main/recipes_source/distributed_checkpoint_recipe.rst>`__
+to show how this can be integrated quite easily with `torch.distributed.checkpoint.async_save`.
+
+
+Notes on Asynchronous Checkpointing
+------------------------------------
+Before getting started with Asynchronous Checkpointing, it's important that we discuss some differences and limitations as compared to synchronous checkpointing.
+Speciically:
+
+* Memory requirements - Asynchronous checkpointing works by first copying models into internal CPU-buffers.
+    This is helpful since it ensures model and optimizer weights are not changing while the model is still checkpointing,
+    but does raise CPU memory by a factor of checkpoint size times the number of process on the host.
```

Comment:
I feel, this `checkpoint size times the number of process on the host` is ambiguous. 1) checkpoint size, is this the global checkpoint size or per-GPU checkpoint size? It is actually the per-GPU checkpoint size. 2) the number of process is also ambiguous. Can we just say the number of GPUs per host?